src/format/xVecLibBoomerangFormatV0_1/GeneralConfig.tsx

```diff
@@ -0,0 +1,157 @@
+        <Box className="Panel-FlexBox">
+          <ObserverInput
+            label="Chassis Name"
+            getValue={() => config.chassisName}
+            setValue={(value: string) => {
+              app.history.execute(`Change chassis variable name`, new UpdateProperties(config, { chassisName: value }));
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => candidate !== ""}
+            sx={{ marginTop: "16px" }}
+          />
+          <ObserverInput
+            label="Movement Timeout"
+            getValue={() => config.movementTimeout.toString()}
+            setValue={(value: string) => {
+              const parsedValue = parseInt(Int.parse(new CodePointBuffer(value))!.value);
+              app.history.execute(
+                `Change default movement timeout to ${parsedValue}`,
+                new UpdateProperties(config, { movementTimeout: parsedValue })
+              );
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => Int.parse(new CodePointBuffer(candidate)) !== null}
+            sx={{ marginTop: "16px" }}
+            numeric
+          />
+        </Box>
+        <Box className="Panel-FlexBox">
+          <ObserverCheckbox
+            label="Use Relative Coordinates"
+            checked={config.relativeCoords}
+            onCheckedChange={value => {
+              app.history.execute(
+                `Set using relative coordinates to ${value}`,
+                new UpdateProperties(config, { relativeCoords: value })
+              );
+            }}
+          />
+        </Box>
+        <Box className="Panel-FlexBox" sx={{ marginTop: "32px" }}>
+          <Button variant="contained" title={`Copy Generated Code (${hotkey})`} onClick={onCopyCode}>
+            Copy Code
+          </Button>
+        </Box>
+      </Box>
+    </>
+  );
+});
+
+// observable class
+export class GeneralConfigImpl implements GeneralConfig {
+  @IsPositive()
+  @Expose()
+  robotWidth: number = 12;
+  @IsPositive()
+  @Expose()
+  robotHeight: number = 12;
+  @IsBoolean()
+  @Expose()
+  robotIsHolonomic: boolean = false;
+  @IsBoolean()
+  @Expose()
+  showRobot: boolean = false;
+  @ValidateNumber(num => num > 0 && num <= 1000) // Don't use IsEnum
+  @Expose()
+  uol: UnitOfLength = UnitOfLength.Inch;
+  @IsPositive()
+  @Expose()
+  pointDensity: number = 2; // inches
+  @IsPositive()
+  @Expose()
+  controlMagnetDistance: number = 5 / 2.54;
+  @Type(() => FieldImageSignatureAndOrigin)
+  @ValidateNested()
+  @IsObject()
+  @Expose()
+  fieldImage: FieldImageSignatureAndOrigin<FieldImageOriginType> =
+    getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+  @IsString()
+  @MinLength(1)
+  @Expose()
+  chassisName: string = "chassis";
+  @ValidateNumber(num => num >= 0)
+  @Expose()
+  movementTimeout: number = 5000;
+  @IsBoolean()
+  @Expose()
+  relativeCoords: boolean = true;
```

Comment:
Are you using `relativeCoords`?

Reply:
Currently its allowed, but support is deprecated.

Reply:
It seems like you are not using it in the `exportCode` function.

Reply:
You can remove it, or I can if you want.

Reply:
You need to remove it. In addition, you need to write test cases for this format.